grpc/src/main/java/com/linecorp/armeria/server/grpc/FramedGrpcService.java

```diff
@@ -320,9 +321,16 @@ private <I, O> void startCall(ServerMethodDefinition<I, O> methodDef, ServiceReq
         call.setListener(listener);
         call.startDeframing();
         ctx.whenRequestCancelling().handle((cancellationCause, unused) -> {
-            final Status status = call.exceptionHandler().apply(ctx, null, cancellationCause, headers);
+            Metadata metadata = Status.trailersFromThrowable(cancellationCause);
+            if (metadata == null) {
+                metadata = new Metadata();
+            }
+            Status status = Status.fromThrowable(cancellationCause);
+            if (status.getCause() != null) {
+                status = call.exceptionHandler().apply(ctx, status, cancellationCause, metadata);
```

Comment:
On the server-side, we passed `status.getCause()` to `exceptionHandler.apply()`. Should we employ the same approach here?

Reply:
Yes, It's fixed. 😉